core-active-state.html

```diff
@@ -0,0 +1,149 @@
+<!--
+Copyright (c) 2015 The Polymer Project Authors. All rights reserved.
+This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+Code distributed by Google as part of the polymer project is also
+subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+-->
+<link rel="import" href="../polymer/polymer.html">
+<dom-module id="core-active-state"></dom-module>
+<script>
+  Polymer({
+    is: 'core-active-state',
+
+    properties: {
+
+      /**
+       * If true, the element is currently active either because the
+       * user is touching it, or the button is a toggle
+       * and is currently in the active state.
+       *
+       * @attribute active
+       * @type boolean
+       * @default false
+       */
+      active: {
+        type: Boolean,
+        notify: true,
+        readOnly: true,
+        observer: 'activeChanged',
+        value: false
+      },
+
+      /**
+       * If true, the user is currently holding down the button.
+       *
+       * @attribute pressed
+       * @type boolean
+       * @default false
+       */
+      pressed: {
+        type: Boolean,
+        value: function() {
+          this._setPressed(this.pressed || false);
+        },
+        readOnly: true,
+        observer: 'pressedChanged'
+      },
+
+      /**
+       * If true, the button toggles the active state with each tap.
+       * Otherwise, the button becomes active when the user is holding
+       * it down.
+       *
+       * @attribute toggle
+       * @type boolean
+       * @default false
+       */
+      toggle: {
+        type: Boolean,
+        value: false
+      },
+
+      /**
+       * If true, the user cannot interact with this element.
+       *
+       * @attribute disabled
+       * @type boolean
+       * @default false
+       */
+      disabled: {
+        type: Boolean,
+        observer: 'disabledChanged',
+        value: false
+      },
+
+      boundDownHandler: {
+        value: function() {
+          return this.onTargetDown.bind(this);
+        }
+      },
+
+      boundUpHandler: {
+        value: function() {
+          return this.onTargetUp.bind(this);
```

Comment:
In case anyone else comes across this discussion, @cdata pointed out that `_setPressed` is not guaranteed to be created at this point and so trying to call `bind` on it might throw.